packages/transformers/js/core/src/lib.rs

```diff
@@ -566,8 +566,9 @@ fn emit(
         None
       },
     ));
-    let config =
-      swc_core::ecma::codegen::Config::default().with_target(swc_core::ecma::ast::EsVersion::Es5);
+    let config = swc_core::ecma::codegen::Config::default()
+      .with_target(swc_core::ecma::ast::EsVersion::Es5)
+      .with_ascii_only(true);
```

Comment:
Does this also affect pre-existing unicode characters? E.g. does it turn `'🙂'` into `'\uD83D\uDE42'`?

Reply:
What changed in 2.7 that caused this issue in the first place?

Reply:
Probably a swc upgrade.
Input: `["\u0085", "\u200b", "\ufffe", "🙂"];`
With 2.6.2: `["\x85", "\u200B", "\uFFFE", "\uD83D\uDE42"]`
With 2.7.0: `["\x85", "​", "￾", "\uD83D\uDE42"]`
With 2.9.0: `["\x85", "​", "￾", "\uD83D\uDE42"]`

So your concern is already the case, it just doesn't happen for single-codepoint characters (or something like that)

> does it turn `'🙂'` into `'\uD83D\uDE42'`?

Reply:
So yeah, this PR would restore the behavior from 2.6.2, which is perfectly fine to me and hasn't bothered anyone since.